### DISCUSSION TITLE
Why should we use pseudo class when we already have AvaloniaProperty?

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#12390] [Q&A]
By what I know, pseudo classes are used for styling, that is, [marking the control with some pseudo class so template author can use selector for certain pseudo classes](https://docs.avaloniaui.net/docs/next/concepts/pseudo-classes#usage). 
But that can also be done by adding some AvaloniaProperty to the control. In Style, [the Selector also support matching properties](https://docs.avaloniaui.net/docs/styling/selectors#propertyequals). Its even more powerful as its not limited to boolean.
So why does people use pseudo classes instead of properties?

**Comment by timunie** [comment]
pseudo classes are a convenient way to express most often needed control status, such as :pointer-over and :disabled which is way shorter to type. It's inspired by CSS. But you are right, Avalonia has may ways to achieve the same. Pick whatever suits you best.
